blender/LilySurfaceScraper/Scrapers/AbstractScraper.py

```diff
@@ -166,12 +176,73 @@ def clearString(self, s):
         return ''.join(filter(lambda x: x in printable, s))
 
     def fetchVariantList(self, url):
+        # **must have self.asset_name**
+
+        # get asset name and variants
+        variants = self._fetchVariantList(url)
+        assetName = self.asset_name
+
+        root = self.getTextureDirectory(os.path.join(self.home_dir, assetName))
+
+        # download thumbnail and make metadata file if meta file is not present
+        metadataFile = os.path.join(root, ".meta")
+        if not os.path.isfile(metadataFile):
+            thumbnailName = self.downloadThumbnail(root, assetName)
+
+            metadata = {
+                "name": assetName,
+                "scraper": self.__class__.__name__,
+                "fetchUrl": url,
+                "thumbnail": thumbnailName,
+                "variants": variants,
+            }
+            with open(metadataFile, "w") as f:
+                json.dump(metadata, f, indent=4)
+        return variants
+
+    def downloadThumbnail(self, assetPath, assetName):
+        thumbnailUrl = self.getThumbnail(assetName)
+        ext = None
+        if thumbnailUrl is None:
+            print("no thumbnail found, not downloading")
+        else:
+            thumbnailReq = self._fetch(thumbnailUrl)
+            if thumbnailReq is None:
+                return None
+            thumbnailType = thumbnailReq.headers["Content-Type"]
+            if thumbnailType == 'image/png':
+                ext = "png"
+            elif thumbnailType == "image/jpeg":
+                ext = "jpg"
+            else:
+                print(f"thumbnail type '{thumbnailType}' is not a valid type")
+
+        if ext is None:
+            return None
+        thumbnailName = f"thumb.{ext}"
+        with open(os.path.join(assetPath, thumbnailName), "wb") as f:
+            f.write(thumbnailReq.content)
+        return thumbnailName
+
+    def _fetchVariantList(self, url):
```

Comment:
Use a different name for what you called `fetchVariantList`, underscore is reserved to private methods.

Reply:
`_fetchVariantList` is a private method that is called by `fetchVariantList`.
`fetchVariantList` gets called by other places and handles downloading the metadata and thumbnail so that scrapers only have to implement `_fetchVariantList`. 
I'll change the name though